rust/src/treefile.rs

```diff
@@ -1153,6 +1187,75 @@ pub(crate) struct LegacyTreeComposeConfigFields {
     pub(crate) automatic_version_prefix: Option<String>,
 }
 
+#[derive(Serialize, Deserialize, Debug, Default, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) struct DeriveCustom {
+    pub(crate) url: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) description: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) struct DeriveInitramfs {
+    pub(crate) regenerate: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) etc: Option<Vec<String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) args: Option<Vec<String>>,
```

Comment:
Not totally sure if this is necessary vs using the existing `initramfs-args`.

Reply:
Hmm, not sure I follow. Aren't these the user-supplied dracut args from `rpm-ostree initramfs --arg`. They're then appended on top of the built-in `initramfs-args` list in the commit metadata.